### DISCUSSION TITLE
File based custom instructions

### DISCUSSION THREAD

**Discussion by BitByBen** [cline/cline#361] [Feature Requests]
Would be handy to have file based instructions to allow per project custom instructions/context similar to .cursorrules

**Comment by ichoosetoaccept** [comment]
Currently Claude Dev's custom instructions are global and you can only set them once. If you switch to a different codebase with completely different requirements, you have to delete what's in the box and supply something new. And do this every time you switch contexts.
I run into this all the time. Would love to see it added.

**Comment by genkernel** [comment]
How do I upvote this x10 times

**Reply by genkernel** [reply]
Im wondering whether Cline can add this feature (or not yet).

**Comment by theviktor** [comment]
this can be a dot file on the directory , like cursor or .gitignore does instructions per directory

**Comment by genkernel** [comment]
> Would be handy to have file based instructions to allow per project custom instructions/context similar to .cursorrules

Just noting similar functionality in aider with [convention files](https://aider.chat/docs/usage/conventions.html#always-load-conventions): https://aider.chat/docs/usage/conventions.html#always-load-conventions

**Comment by soctib** [comment]
Where are the custom instructions stored anyway? I was hoping they were just put in to the VSCode settings, because they can be per user or per workspace - problem solved.

**Comment by lloydchang** [comment]
as an alternative, you can try a fork https://github.com/RooVetGit/Roo-Cline/commit/a6bfbf8bcb4b27ae02a8ae35c4b0fdd5a5417416#diff-ec5bfb97fa984921dcb420c3fcf9747854c3302ac5ec437f1fe604ee68fc4e7aR287
> `const ruleFiles = ['.clinerules', '.cursorrules']`

and its `.vsix` at https://marketplace.visualstudio.com/items?itemName=RooVeterinaryInc.roo-cline

from https://discord.com/channels/1275535550845292637/1275535550845292640/1312113711842066524

@mrubens wrote:
> In the fork I've been working on I have Cline reading .cursorrules and .clinerules from the working directory and interpolating them into the system prompt if present (at least in our use case most of the things we tell Cursor are also relevant for Cline). 

related https://github.com/cline/cline/discussions/361 https://github.com/cline/cline/discussions/408 https://github.com/cline/cline/discussions/622

**Comment by saoudrizwan** [comment]
You can now specify custom instructions for your project by adding a root-level `.clinerules` file. This allows you to define project-specific behaviors and guidelines for Cline–perfect for setting conventions, pointing to important documentation, or providing context about your project's architecture.
